client/signup/steps/woocommerce-install/confirm/index.tsx

```diff
@@ -0,0 +1,178 @@
+import { getSiteDomain } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import type { GoToStep } from '../../../types';
+import type { AppState } from 'calypso/types';
+
+import './style.scss';
+
+interface Props {
+	goToStep: GoToStep;
+	stepSectionName: string;
+	isReskinned: boolean;
+	headerTitle: string;
+	headerDescription: string;
+}
+
+export default function Confirm( {
+	goToStep,
+	isReskinned,
+	stepSectionName,
+	headerTitle,
+	headerDescription,
+}: Props ): ReactElement | null {
+	const { __ } = useI18n();
+	const siteId = useSelector( getSelectedSiteId ) as number;
+	const dispatch = useDispatch();
+
+	// Request eligibility data.
+	useEffect( () => {
+		if ( ! siteId ) {
+			return;
+		}
+		dispatch( fetchAutomatedTransferStatusOnce( siteId ) );
+		dispatch( requestEligibility( siteId ) );
+	}, [ siteId, dispatch, goToStep ] );
+
+	// Check whether it's requesting eligibility data.
+	const isFetchingTransferStatus = !! useSelector( ( state ) =>
+		isFetchingAutomatedTransferStatus( state, siteId )
+	);
+
+	// Get eligibility data.
+	const {
+		eligibilityHolds,
+		eligibilityWarnings: allEligibilityWarnings,
+	}: EligibilityData = useSelector( ( state ) => getEligibility( state, siteId ) );
+
+	// Check whether the wpcom.com subdomain warning is present.
+	const wordPressSubdomainWarning = allEligibilityWarnings?.find(
+		( { id } ) => id === 'wordpress_subdomain'
+	);
+
+	const eligibilityWarnings = allEligibilityWarnings?.filter(
+		( { id } ) => id !== 'wordpress_subdomain'
+	);
+
+	// Pick the wpcom subdomain.
+	const wpcomDomain = useSelector( ( state: AppState ) => getSiteDomain( state, siteId ) );
+
+	const isProcessing = ! siteId || isFetchingTransferStatus || ! wordPressSubdomainWarning;
+
+	function getWPComSubdomainWarningContent() {
+		// Get staging sudomain based on the wpcom subdomain.
+		const stagingDomain = wpcomDomain?.replace( /\b.wordpress.com/, '.wpcomstaging.com' );
+
+		return (
+			<>
+				<div className="confirm__image-container">
+					{ isProcessing && <LoadingEllipsis /> }
+					<img src={ yourNewStoreImage } alt="" />
+				</div>
+				<div className="confirm__instructions-container">
+					<div className="confirm__instructions-title confirm__instructions-wpcom-domain">
+						{ wpcomDomain }
+					</div>
+
+					<div className="confirm__instructions-title">{ stagingDomain }</div>
+
+					<p>
+						{ __(
+							'By installing this product your subdomain will change. You can change it later to a custom domain and we will pick up the tab for a year.'
+						) }
+					</p>
+
+					<p>
+						{ createInterpolateElement( __( '<a>Contact support</a> for help and questions.' ), {
+							a: <a href="#support-link" />,
+						} ) }
+					</p>
+
+					<NextButton
+						disabled={ isProcessing }
+						onClick={ () => {
+							if ( eligibilityHolds?.length || eligibilityWarnings?.length ) {
+								return goToStep( 'confirm', 'eligibility_substep' );
```

Comment:
I just want to understand what is a substep? specifically what happens when we call `goToStep` with an `elegibility_substep`. It will render this same component but with some props passed and we will do some kind of conditional render?

Reply:
`eligibility_substep` is a step section name, it'll render the 'confirm' step with a [sectionName](https://github.com/Automattic/wp-calypso/blob/97795def009da339bdfcda672c19d2d467a1355e/client/signup/controller.js#L307) in the url and passed in as a prop.